examples/distributed_tree/distributed_intersects.cpp

```diff
@@ -0,0 +1,106 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <cstdarg>
+#include <cstdio>
+#include <iostream>
+#include <random>
+#include <vector>
+
+#include <mpi.h>
+
+using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+using MemorySpace = ExecutionSpace::memory_space;
+
+namespace Example
+{
+template <class Points>
+struct Intersects
+{
+  Points points;
+  float radius;
+};
+template <class Points>
+Intersects(Points const &, int) -> Intersects<Points>;
```

Comment:
```suggestion
Intersects(Points, int) -> Intersects<Points>;
```
Wondering whether we should comment that we don't need it from C++20